File: app/src/main/java/com/mikepenz/fastadapter/app/adapters/SampleWrapAdapter.java
package com.mikepenz.fastadapter.app.adapters;

import com.mikepenz.fastadapter.IItem;
import com.mikepenz.fastadapter.commons.adapters.AbstractWrapAdapter;

import java.util.List;

/**
 * Created by mikepenz on 03.03.16.
 */
public class SampleWrapAdapter<Item extends IItem> extends AbstractWrapAdapter<Item> {

    public SampleWrapAdapter(List<Item> list) {
        super(list);
    }

    @Override
    public boolean shouldInsertItemAtPosition(int position) {
        return position == 10;
    }

    @Override
    public int itemInsertedBeforeCount(int position) {
        if (position > 10) {
            return 1;
        }
        return 0;
    }
}

File: app/src/main/java/com/mikepenz/fastadapter/app/adapters/SampleWrapAdapter.kt
package com.mikepenz.fastadapter.app.adapters

import androidx.recyclerview.widget.RecyclerView

import com.mikepenz.fastadapter.IItem
import com.mikepenz.fastadapter.adapters.AbstractWrapAdapter

/**
 * Created by mikepenz on 03.03.16.
 */
class SampleWrapAdapter<Item : IItem<VH>, VH : RecyclerView.ViewHolder>(list: List<Item>) : AbstractWrapAdapter<Item, VH>(list) {

    override fun shouldInsertItemAtPosition(position: Int): Boolean {
        return position == 10
    }

    override fun itemInsertedBeforeCount(position: Int): Int {
        return if (position > 10) {
            1
        } else 0
    }
}
